calico/felix/endpoint.py

```diff
@@ -51,11 +51,14 @@ def __init__(self, config, ip_type,
         self.dispatch_chains = dispatch_chains
         self.rules_mgr = rules_manager
 
-        # Endpoints that are in use; LocalEndpoint actors referenced by
-        # endpoint_id.
+        # All endpoint dicts that we know about.
         self.endpoints_by_id = {}
         self.endpoint_id_by_iface_name = {}
 
+        # Set of endpoints that are live on this host.  I.e. ones that we've
+        # increffed.
+        self.local_endpoint_ids = set()
```

Comment:
Tracking this explicitly to make it much simpler to get the code correct (there was a bug before)